File: solutions/Permutation-Sequence/Permutation-Sequence.cpp

class Solution {
public:
    string getPermutation(int n, int k) {
        string ans="";
        vector<int> fac(n+1,1);
        vector<char> num(n,'1');
        for(int i=2;i<=n;i++)
        {
            fac[i]=fac[i-1]*i;
            num[i-1]=i+'0';
        }
        k--;
        for(int i=n-1;i>=0;i--)
        {
            int cur=k/fac[i];
            k%=fac[i];
            ans+=num[cur];
            num.erase(num.begin()+cur);
        }
        return ans;
    }
};


 

File: solutions/Permutation-Sequence/Permutation-Sequence.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function (n, k) {
  let ans = '';
  const fac = Array.from({ length: n + 1 }, () => 1);
  for (let i = 1; i <= n; i++) {
    fac[i] = fac[i - 1] * i;
  }
  const nums = Array.from({ length: n }, (a, i) => i + 1);
  k--;

  for (let i = 1; i <= n; i++) {
    const index = Math.floor(k / fac[n - i]);
    ans += String.fromCharCode(nums[index] + '0'.charCodeAt(0));
    nums.splice(index, 1);
    k -= index * fac[n - i];
  }

  return ans;
};